src/charm.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Charmed Kubernetes Operator for Apache Spark History Server."""
+
+import logging
+
+from ops.charm import CharmBase, InstallEvent
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+
+from config import SparkHistoryServerConfigModel
+from constants import (
+    CONFIG_KEY_S3_LOGS_DIR,
+    CONTAINER,
+    CONTAINER_LAYER,
+    SPARK_HISTORY_SERVER_LAUNCH_CMD,
+    SPARK_PROPERTIES_FILE,
+    SPARK_USER,
+    SPARK_USER_GID,
+    SPARK_USER_GROUP,
+    SPARK_USER_UID,
+    SPARK_USER_WORKDIR,
+)
+
+# Log messages can be retrieved using juju debug-log
+logger = logging.getLogger(__name__)
+
+VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
```

Comment:
**question** we already have some levels in the utils module, don't we? can we use those?